agent/utils/utils.go

```diff
@@ -129,6 +130,7 @@ func parseRequestIDs(response *http.Response) ([]string, error) {
 		return nil, fmt.Errorf("Failed to read the forwarded request: %q\n", err.Error())
 	}
 	if response.StatusCode != http.StatusOK {
+		metricHandler.WriteResponseCodeMetric(response.StatusCode)
```

Comment:
Should we be emitting metrics for internal agent requests? These don't correspond to user requests, and in fact for list pending requests they often time out and fail under normal conditions if there's no traffic.

Reply:
If there's no potential implications for the user, then I think it does make sense to remove the call here.